Compile error due to non-virtual destructor

```
/usr/bin/clang++-7  -DGFLAGS_IS_A_DLL=0 -D__CLANG_SUPPORT_DYN_ANNOTATION__ -I../third_party/prjxray/third_party/abseil-cpp -Ithird_party/prjxray/third_party/gflags/include -I../third_party/prjxray/lib/include -I../third_party/prjxray/third_party/yaml-cpp/include -O3 -DNDEBUG   -Wall -Werror -std=gnu++14 -MD -MT third_party/prjxray/tools/CMakeFiles/xc7frames2bit.dir/xc7frames2bit.cc.o -MF third_party/prjxray/tools/CMakeFiles/xc7frames2bit.dir/xc7frames2bit.cc.o.d -o third_party/prjxray/tools/CMakeFiles/xc7frames2bit.dir/xc7frames2bit.cc.o -c ../third_party/prjxray/tools/xc7frames2bit.cc                                        
In file included from ../third_party/prjxray/tools/xc7frames2bit.cc:4:
In file included from ../third_party/prjxray/lib/include/prjxray/xilinx/architectures.h:4:
In file included from ../third_party/prjxray/third_party/abseil-cpp/absl/types/variant.h:77:
../third_party/prjxray/third_party/abseil-cpp/absl/types/internal/variant.h:1302:7: error: destructor called on non-final 'prjxray::xilinx::Series7' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
      variant_internal::AccessUnion(self->state_, i).~Alternative();
      ^
../third_party/prjxray/third_party/abseil-cpp/absl/base/internal/invoke.h:152:12: note: in instantiation of function template specialization 'absl::variant_internal::VariantStateBaseDestructorNontrivial<prjxray::xilinx::Series7>::Destroyer::operator()<0>' requested here
    return std::forward<F>(f)(std::forward<Args>(args)...);
           ^
../third_party/prjxray/third_party/abseil-cpp/absl/base/internal/invoke.h:179:37: note: in instantiation of function template specialization 'absl::base_internal::Callable::Invoke<absl::variant_internal::VariantStateBaseDestructorNontrivial<prjxray::xilinx::Series7>::Destroyer, std::integral_constant<unsigned long, 0> >' requested here                                                                                    
  return Invoker<F, Args...>::type::Invoke(std::forward<F>(f),
                                    ^
../third_party/prjxray/third_party/abseil-cpp/absl/types/internal/variant.h:294:33: note: in instantiation of function template specialization 'absl::base_internal::Invoke<absl::variant_internal::VariantStateBaseDestructorNontrivial<prjxray::xilinx::Series7>::Destroyer, std::integral_constant<unsigned long, 0> >' requested here                                                                                            
    return absl::base_internal::Invoke(absl::forward<Op>(op), SizeT<I>());
                                ^
../third_party/prjxray/third_party/abseil-cpp/absl/types/internal/variant.h:359:43: note: in instantiation of member function 'absl::variant_internal::ReachableSwitchCase<absl::variant_internal::VariantStateBaseDestructorNontrivial<prjxray::xilinx::Series7>::Destroyer, 0>::Run' requested here
        return PickCase<Op, 0, EndIndex>::Run(absl::forward<Op>(op));
                                          ^
../third_party/prjxray/third_party/abseil-cpp/absl/types/internal/variant.h:1312:48: note: in instantiation of function template specialization 'absl::variant_internal::VisitIndicesSwitch<1>::Run<absl::variant_internal::VariantStateBaseDestructorNontrivial<prjxray::xilinx::Series7>::Destroyer>' requested here
  void destroy() { VisitIndices<sizeof...(T)>::Run(Destroyer{this}, index_); }
                                               ^
../third_party/prjxray/third_party/abseil-cpp/absl/types/internal/variant.h:1314:45: note: in instantiation of member function 'absl::variant_internal::VariantStateBaseDestructorNontrivial<prjxray::xilinx::Series7>::destroy' requested here
  ~VariantStateBaseDestructorNontrivial() { destroy(); }
                                            ^
../third_party/prjxray/third_party/abseil-cpp/absl/types/internal/variant.h:1327:15: note: in instantiation of member function 'absl::variant_internal::VariantStateBaseDestructorNontrivial<prjxray::xilinx::Series7>::~VariantStateBaseDestructorNontrivial' requested here
  using Base::Base;
              ^
../third_party/prjxray/third_party/abseil-cpp/absl/types/internal/variant.h:1302:55: note: qualify call to silence this warning                                                                                   
      variant_internal::AccessUnion(self->state_, i).~Alternative();
                                                      ^
                                                      prjxray::xilinx::Series7::
1 error generated.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,12 @@ add_subdirectory(third_party/gflags EXCLUDE_FROM_ALL)
 add_subdirectory(third_party/cctz EXCLUDE_FROM_ALL)
 add_subdirectory(third_party/abseil-cpp EXCLUDE_FROM_ALL)
 
+include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-Wundefined-var-template" CXX_COMPILER_SUPPORTS_UNDEF_VAR)
+if(${CXX_COMPILER_SUPPORTS_UNDEF_VAR})
+    add_compile_options("-Wno-undefined-var-template")
+endif()
+
 option(YAML_CPP_BUILD_TESTS "" OFF)
 add_subdirectory(third_party/yaml-cpp EXCLUDE_FROM_ALL)
 target_include_directories(yaml-cpp PUBLIC

--- a/lib/include/prjxray/xilinx/architectures.h
+++ b/lib/include/prjxray/xilinx/architectures.h
@@ -19,6 +19,7 @@ class Architecture {
        public:
 	using Container = absl::variant<Series7>;
 	virtual const std::string& name() const = 0;
+	virtual ~Architecture() {}
 };
 
 class Series7 : public Architecture {

